test/tests/neutronics/heat_source/particles_from_postprocessor.i

```diff
@@ -0,0 +1,73 @@
+[Mesh]
+  [sphere]
+    type = FileMeshGenerator
+    file = ../meshes/sphere.e
+  []
+  [solid]
+    type = CombinerGenerator
+    inputs = sphere
+    positions = '0 0 0
+                 0 0 4
+                 0 0 8'
+  []
+  [solid_ids]
+    type = SubdomainIDGenerator
+    input = solid
+    subdomain_id = '100'
+  []
+  [fluid]
+    type = FileMeshGenerator
+    file = stoplight.exo
+  []
+  [fluid_ids]
+    type = SubdomainIDGenerator
+    input = fluid
+    subdomain_id = '200'
+  []
+  [combine]
+    type = CombinerGenerator
+    inputs = 'solid_ids fluid_ids'
+  []
+[]
+
+[Functions]
+  [nparticles_fn]
+    type = ParsedFunction
+    value = 100*t
+  []
+[]
+
+[Postprocessors]
+  [n_particles]
+    type = FunctionValuePostprocessor
+    function = nparticles_fn
+    execute_on = 'initial timestep_begin'
```

Comment:
Do you need the `initial` here? The simpler the better in terms of a new user copying this as a starting point